packages/grafana-prometheus/src/datasource.test.ts

```diff
@@ -1093,6 +1097,18 @@ describe('PrometheusDatasource2', () => {
         const results = await runAnnotationQuery([[2 * 60000], [1]]);
         expect(results.map((result) => [result.time, result.timeEnd])).toEqual([[120000, 120000]]);
       });
+
+      it('should handle non-default step in executedQuery', async () => {
```

Comment:
This is looking good. Can you make this test show that two different steps produce different results? One result with the standard step will show individual annotation and one result with a greater step with group the annoations.

Then update the test description to say something like "should group annotations over wider range when the step grows larger"

```
Smaller range
1m0s => [
          [120000, 120000],
          [240000, 240000],
          [600000, 600000],
          [720000, 720000],
        ]

2m0s => [
          [120000, 240000],
          [600000, 720000],
        ]

Reply:
Good idea! Have tried to fix that.